src/api/controller/getListingDetails.ts

```diff
@@ -0,0 +1,107 @@
+import {
+	EventTracker,
+	ListingDetails,
+	ListingQueryObject,
+	ListingTimeline,
+	Request,
+} from "@/src/types";
+import { FastifyReply } from "fastify";
+import getTokenIdArray from "@/src/api/utils/getTokenIdArray";
+import { logger } from "@/src/logger";
+
+export default async function getListingDetails(
+	request: Request,
+	reply: FastifyReply
+): Promise<FastifyReply> {
+	try {
+		const eventTracker = this.mongo.db.collection("EventTracker");
+		const listingId = (request.params as ListingQueryObject).listingId;
+
+		let data = await eventTracker
+			.find({ streamId: listingId })
+			.sort({ version: "asc" });
+		data = await data.toArray();
+
+		if (!data || data.length === 0)
+			return reply.status(500).send({ error: "Listing Not found!" });
+
+		const listingData = data as EventTracker[];
+		let tokenIds = [],
+			listingType,
+			assetId,
+			sellPrice;
+
+		const timeline: ListingTimeline[] = [];
+
+		const listingStarted = listingData.find(
+			(nft) => nft.eventType === "LISTING_STARTED"
+		);
+		if (listingStarted) {
+			const eventDetails = JSON.parse(listingStarted.data);
+
+			tokenIds = eventDetails.tokenIds;
+			assetId = eventDetails.assetId;
+			sellPrice = eventDetails.sellPrice;
```

Comment:
`sellPrice` from Mongo include the symbol, we need to split it up and ensure it's Number type